src/Orleans/Messaging/IncomingMessageBuffer.cs

```diff
@@ -128,6 +131,19 @@ public bool TryDecodeMessage(out Message msg)
             headerLength = 0;
             bodyLength = 0;
 
+            return true;
+        }
+
+        void AdjustBuffer()
```

Comment:
And also to increase readability, in line 153 (of the original version) have the branching logic to how to calculate `backfillBytes`, but have the call to readBuffer.AddRange only once. And update the comment to: "restore buffer up to DEFAULT_MAX_SUSTAINED_RECEIVE_BUFFER_SIZE" (instead of shrink, which is correct from the high level perspective of the method, but at this step in particular you are still expanding/restoring it)